pkg/common/path.go

```diff
@@ -80,6 +88,54 @@ func ExpandPackageDir(f genericclioptions.FileNameFlags) (genericclioptions.File
 	return f, nil
 }
 
+// FilterInputFile copies the resource config on stdin into a file
+// at the tmpDir, filtering the inventory object. It is the
+// responsibility of the caller to clean up the tmpDir. Returns
+// an error if one occurs.
+func FilterInputFile(in io.Reader, tmpDir string) error {
+	// Copy the config from "in" into a local temp file.
+	dir, err := ioutil.TempDir("", tmpDirPrefix)
+	if err != nil {
+		return err
+	}
+	tmpFile, err := ioutil.TempFile(dir, fileRegexp)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(dir)
+	klog.V(6).Infof("Temp File: %s", tmpFile.Name())
+	if _, err := io.Copy(tmpFile, in); err != nil {
+		return err
+	}
+	// Read the config stored locally, parsing into RNodes
+	r := kio.LocalPackageReader{PackagePath: dir}
```

Comment:
It's unfortunate that this type can't read from a stream. It looks like [it intends to walk a whole tree](https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/kio/pkgio_reader.go#L171-L232), though, so it also accepting a stream would be limiting a lot of the work it tries to do.